ClassLibrary/ClassLibrary/RepositoryBooks.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace ClassLibrary
+{
+    public class RepositoryBooks
+    {
+        // looks like with "DepartmentReaders", you can write for one occasion...  
+
+        //for comfortable I do it open, but I break AV1014 (An object should not expose any other classes it depends on for Library)
+        private List<Book> Books
+        {
+            get;
+            set;
+        }
+            = new List<Book>();
+
+        public void AddBook(Book book)
+        {
+            if (book == null)
+            {
+                throw new ArgumentNullException("Book's reference is null");
+            }
+
+            Book foundBook = Books?.FirstOrDefault(x => x.ISBN == book.ISBN);
+            if (foundBook != null)
+            {
+                throw new ArgumentException($"This book's [{book.ISBN},{book.NameBook}] exists");
+            }
+            else
+            {
+                Books.Add(book);
+            }
+        }
+
+ 
+        public List<Book> FindBooks(String NameBook="",String NameAutor="",String LastNameAutor="")
+        {
+            IEnumerable < Book > request = Books;
+
+            if (NameBook.IsNullOrWhiteSpace()==false&&request?.Count()!=0)
```

Comment:
1. Create an extension for `string` that will be the opposite of `IsNullOrWhiteSpace()` and use it here.
2. you don't need to check for `request.Count()`. and it iterates through the collection (`request`) every time, so affects performance.

Reply:
3. the new code looks excellent, right?